DNN Platform/Library/Security/PortalSecurity.cs

```diff
@@ -809,19 +809,43 @@ public bool ValidateInput(string userInput, FilterFlag filterType)
 
             return (userInput == filteredInput);
         }
-		
-		#endregion
-		
-		#region Public Shared/Static Methods
+
+        /// <summary>
+        /// This function loops through every portal that has set its own AllowedExtensionWhitelist
+        /// and checks that there are no extensions there that are restriced by the host
+        /// 
+        /// The only time we should call this is if the host allowed extensions list has changed
+        /// </summary>
+        public void CheckAllPortalFileExtensionWhitelists()
```

Comment:
If this is internal, can we mark this internal to prevent external callers?

Reply:
Additionally I think we might want a bit stronger name on this, as this isn't a check, it actually updates/changes data if necessary